crates/core/src/download.rs

```diff
@@ -106,16 +106,12 @@ pub fn find_install_path_sync(dependency: &Dependency, deps: impl AsRef<Path>) -
     let Ok(read_dir) = fs::read_dir(deps.as_ref()) else {
         return None;
     };
-    for entry in read_dir {
-        let Ok(entry) = entry else {
-            continue;
-        };
-        let path = entry.path();
-        if install_path_matches(dependency, &path) {
-            return Some(path);
-        }
-    }
-    None
+
+    read_dir
+        .par_bridge()
+        .into_par_iter()
```

Comment:
Not convinced this is beneficial here as there are a limited number of top-level dependencies in any given project. I would be worried that the rayon overhead negates any potential gains from parallelizing iteration in a single level of the filesytem. I like the rewrite to iterator chaining however!